ocp_resources/node_network_configuration_policy.py

```diff
@@ -370,32 +370,26 @@ def _process_failed_status(self, failed_condition_reason):
 
         raise NNCPConfigurationFailed(f"Reason: {failed_condition_reason}\n{last_err_msg}")
 
+    @retry(wait_timeout=120, sleep=5)
     def wait_for_status_success(self):
         failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
         no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
 
-        # if we get here too fast there are no conditions, we need to wait.
-        self.wait_for_configuration_conditions_unknown_or_progressing()
-
-        samples = TimeoutSampler(wait_timeout=self.success_timeout, sleep=1, func=lambda: self.status)
-        try:
-            for sample in samples:
-                if sample == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED:
-                    self.logger.info(f"NNCP {self.name} configured Successfully")
-                    return sample
-
-                elif sample == no_match_node_condition_reason:
-                    raise NNCPConfigurationFailed(f"{self.name}. Reason: {no_match_node_condition_reason}")
-
-                elif sample == failed_condition_reason:
-                    self._process_failed_status(failed_condition_reason=failed_condition_reason)
-
-        except (TimeoutExpiredError, NNCPConfigurationFailed):
-            self.logger.error(
-                f"Unable to configure NNCP {self.name} "
-                f"{f'nodes: {[node.name for node in self.nodes]}' if self.nodes else ''}"
-            )
-            raise
+        available_condition = [
+            condition
+            for condition in self.instance.get("status", {}).get("conditions", [])
+            if condition and condition["type"] == self.Conditions.Type.AVAILABLE
+        ]
+        if available_condition:
+            if available_condition[0]["status"] == self.Condition.Status.TRUE:
```

Comment:
`available_condition` is a list, so I guess it can have more than one? 
If so please make sure you operate on the right one and not use [0] by default.

And i this is the case the name should be `available_conditions`

Reply:
No, in status.conditions, there is only one type that is "Available". So it would only be one value in this list - if we find a match.